src/modules/product-category/entities/product-category.entity.ts

```diff
@@ -0,0 +1,25 @@
+import { Entity, PrimaryGeneratedColumn, Column, Index, OneToMany } from 'typeorm';
+import { Product } from '../../products/entities/product.entity';
+
+@Entity('categories')
+export class ProductCategory {
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
```

Comment:
remove this id column from here. Rather extend the AbstractBaseEntity it provides these columns out of the box